model.py

```diff
@@ -154,15 +164,13 @@ def forward(self, input_ids, labels=None):
         x += self.wpe(position_ids)
         x = self.drop_embd(x)
 
-        causal_attention_mask = torch.triu(
-            torch.ones(
-                (sequence_length, sequence_length),
-                dtype=x.dtype,
-                device=x.device,
-            ),
-            diagonal=1,
-        ).unsqueeze(0).unsqueeze(0)
-        causal_attention_mask *= torch.finfo(causal_attention_mask.dtype).min
+        causal_attention_mask = create_broadcasted_autoregressive_mask(
+            batch_size=batch_size,
+            num_heads=1,
```

Comment:
```suggestion
            num_heads=self.config.heads,
```

Reply:
Creating mask explicitly in https://github.com/Cerebras/gigaGPT/pull/5 would be preferable.